sdk/identity/identity/src/credentials/managedIdentityCredential/imdsMsi.ts

```diff
@@ -74,17 +76,20 @@ function prepareRequestOptions(
 }
 
 /**
- * Defines how to determine whether the Azure IMDS MSI is available, and also how to retrieve a token from the Azure IMDS MSI.
+ * Defines how to determine whether the Azure IMDS MSI is available.
+ *
+ * Actually getting the token once we determine IMDS is available is handled by MSAL.
  */
-export const imdsMsi: MSI = {
+export const imdsMsi = {
```

Comment:
We no longer have a concept of `MSI as an interface`

We have:

- imds MSI with `isAvailable` only
- tokenExchangeMsi with `isAvailable` and `getToken`

With that in mind, removing the interface here simplifies the usage and avoids pretending that `getToken` is used in more than one place